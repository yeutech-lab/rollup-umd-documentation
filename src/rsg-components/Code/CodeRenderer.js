import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';

const styles = ({ fontFamily }) => ({
	code: {
		fontFamily: fontFamily.monospace,
		fontSize: 'inherit',
		color: 'inherit',
		background: 'transparent',
		whiteSpace: 'inherit',
	},
});

export function CodeRenderer({ classes, className, children }) {
	const classNames = cx(className, classes.code);

	const isHighlighted = className && className.indexOf('lang-') !== -1;
	if (isHighlighted) {
		return <code className={classNames} title={console.log('test')} dangerouslySetInnerHTML={{ __html: children }} />;
	}
	return <code className={classNames}>{children}</code>;
}
CodeRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	className: PropTypes.string,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(CodeRenderer);
