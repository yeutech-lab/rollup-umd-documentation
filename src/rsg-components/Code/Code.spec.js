import React from 'react';
import { CodeRenderer } from './CodeRenderer';

describe('Code blocks', () => {
	it('should render code', () => {
		const code = '<button>OK</button>';
		const actual = shallow(<CodeRenderer classes={{}}>{code}</CodeRenderer>);

		expect(actual).toMatchSnapshot();
	});

	it('should render code to be highlighted', () => {
		const code = '<button>OK</button>';
		const actual = render(
			<CodeRenderer classes={{}} className="lang-html">
				{code}
			</CodeRenderer>
		);

		expect(actual).toMatchSnapshot();
	});
});
