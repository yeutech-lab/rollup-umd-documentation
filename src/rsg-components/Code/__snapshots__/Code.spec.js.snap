// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code blocks should render code 1`] = `
<code
  className=""
>
  &lt;button&gt;OK&lt;/button&gt;
</code>
`;

exports[`Code blocks should render code to be highlighted 1`] = `
<code
  class="lang-html"
>
  <button>
    OK
  </button>
</code>
`;
